src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -59,12 +59,36 @@ public FeatureDiagnosticData[] SetupFeatures(FeatureConfigurationContext feature
 
         public async Task StartFeatures(IServiceProvider builder, IMessageSession session, CancellationToken cancellationToken = default)
         {
+            var startedTaskControllers = new List<FeatureStartupTaskController>();
+
             // sequential starting of startup tasks is intended, introducing concurrency here could break a lot of features.
             foreach (var feature in enabledFeatures.Where(f => f.Feature.IsActive))
             {
                 foreach (var taskController in feature.TaskControllers)
                 {
-                    await taskController.Start(builder, session, cancellationToken).ConfigureAwait(false);
+                    try
+                    {
+                        await taskController.Start(builder, session, cancellationToken).ConfigureAwait(false);
+                    }
+                    catch (Exception startException)
+                    {
+                        var exceptions = new List<Exception> { startException };
+
+                        var stop = Task.WhenAll(startedTaskControllers.Select(controller => controller.Stop(cancellationToken)));
```

Comment:
Passing `cancellationToken` to `Stop` was the sticking point, but we came to the conclusion that a feature which abstracts a background task, such as https://github.com/Particular/NServiceBus.Heartbeat, effectively does _not_ support cancellation in its `Stop` method.

The contract is: we will call `Start`, passing a token, and if that token is cancelled, we will then call `Stop`, but we will pass in that same token. `Stop` can then decide if it supports cancellation or not, and honour the token accordingly. And for background task abstractions, `Stop` should _not_ honour the token. I.e. it should _always_ stop the background task.